services/account/apps/api/src/guards/read-only.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { ConfigService } from '#lib/config';
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+
+@Injectable()
+export class ReadOnlyGuard implements CanActivate {
+  // eslint-disable-next-line no-useless-constructor, no-empty-function
+  constructor(private configService: ConfigService) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const { isReadOnly } = this.configService;
+    if (isReadOnly) {
+      // If the system is in read-only mode, disable certain routes
+      const request = context.switchToHttp().getRequest();
+      const { method } = request;
+      const readOnlyMethods = ['POST', 'PUT', 'DELETE']; // Methods that should be restricted
```

Comment:
nit: this name implies the opposite to me, that these are the allowed methods for read-only
```suggestion
      const restrictedMethods = ['POST', 'PUT', 'DELETE']; // Methods that should be restricted
```